src/V3LinkInc.cpp

```diff
@@ -0,0 +1,247 @@
+            whilep->addPrecondsp(newp);
+        } else {
+            nodep->v3fatalSrc("Unknown InsertMode");
+        }
+        m_insMode = IM_AFTER;
+        m_insStmtp = newp;
+    }
+
+    // VISITORS
+    virtual void visit(AstNodeModule* nodep) VL_OVERRIDE {
+        // Reset increments count
+        m_modIncrementsNum = 0;
+        iterateChildren(nodep);
+    }
+    virtual void visit(AstWhile* nodep) VL_OVERRIDE {
+        // Special, as statements need to be put in different places
+        // Preconditions insert first just before themselves (the normal
+        // rule for other statement types)
+        m_insStmtp = NULL;  // First thing should be new statement
+        iterateAndNextNull(nodep->precondsp());
+        // Conditions insert first at end of precondsp.
+        m_insMode = IM_WHILE_PRECOND;
+        m_insStmtp = nodep;
+        iterateAndNextNull(nodep->condp());
+        // Body insert just before themselves
+        m_insStmtp = NULL;  // First thing should be new statement
+        iterateAndNextNull(nodep->bodysp());
+        iterateAndNextNull(nodep->incsp());
+        // Done the loop
+        m_insStmtp = NULL;  // Next thing should be new statement
+    }
+    virtual void visit(AstNodeFor* nodep) VL_OVERRIDE {  // LCOV_EXCL_LINE
+        nodep->v3fatalSrc(
+            "For statements should have been converted to while statements in V3Begin.cpp");
+    }
+    virtual void visit(AstNodeStmt* nodep) VL_OVERRIDE {
+        if (!nodep->isStatement()) {
+            iterateChildren(nodep);
+            return;
+        }
+        m_insMode = IM_BEFORE;
+        m_insStmtp = nodep;
+        iterateChildren(nodep);
+        m_insStmtp = NULL;  // Next thing should be new statement
+    }
+    void unsupported_visit(AstNode* nodep) {
+        m_unsupportedHere = true;
+        UINFO(9, "Marking unsupported " << nodep << endl);
+        iterateChildren(nodep);
+        m_unsupportedHere = false;
+    }
+    virtual void visit(AstLogAnd* nodep) VL_OVERRIDE { unsupported_visit(nodep); }
+    virtual void visit(AstLogOr* nodep) VL_OVERRIDE { unsupported_visit(nodep); }
+    virtual void visit(AstLogEq* nodep) VL_OVERRIDE { unsupported_visit(nodep); }
+    virtual void visit(AstLogIf* nodep) VL_OVERRIDE { unsupported_visit(nodep); }
+    virtual void visit(AstCond* nodep) VL_OVERRIDE { unsupported_visit(nodep); }
+    virtual void visit(AstCondBound* nodep) VL_OVERRIDE { unsupported_visit(nodep); }
+    void prepost_visit(AstNodeTriop* nodep) {
+        // Check if we are underneath a statement
+        if (!m_insStmtp) {
+            prepost_non_stmt_visit(nodep);
+        } else {
+            prepost_stmt_visit(nodep);
+        }
+    }
+    void prepost_non_stmt_visit(AstNodeTriop* nodep) {
+        iterateChildren(nodep);
+
+        AstNode* lhsp = nodep->lhsp();
+        AstNode* rhsp = nodep->rhsp();
+        AstConst* constp = VN_CAST(nodep->thsp(), Const);
```

Comment:
Change lhsp->thsp (preinc), rhsp->lhsp (postinc), thsp->lhsp (const), to match the "shuffling" in verilog.y

One of many:
```suggestion
        AstConst* constp = VN_CAST(nodep->lhsp(), Const);
```